src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -56,12 +59,26 @@ internal class GrpcWorker : IWorker
             _startupOptions = startupOptions ?? throw new ArgumentNullException(nameof(startupOptions));
             _serializer = workerOptions.Value.Serializer ?? throw new InvalidOperationException(nameof(workerOptions.Value.Serializer));
             _inputConversionFeatureProvider = inputConversionFeatureProvider ?? throw new ArgumentNullException(nameof(inputConversionFeatureProvider));
+            _functionMetadataProvider = functionMetadataProvider ?? throw new ArgumentNullException(nameof(functionMetadataProvider));
         }
 
         public async Task StartAsync(CancellationToken token)
         {
             var eventStream = _rpcClient.EventStream(cancellationToken: token);
 
+            if(_startupOptions.Value.AzureWebJobsScriptRoot is null)
+            {
+                throw new ArgumentNullException(nameof(_startupOptions.Value.AzureWebJobsScriptRoot));
+            }
+
+            if (!string.IsNullOrEmpty(_startupOptions.Value.EnableWorkerIndexing))
+            {
+                if (_startupOptions.Value.EnableWorkerIndexing.Equals("true", StringComparison.OrdinalIgnoreCase))
```

Comment:
[This host PR](https://github.com/Azure/azure-functions-host/pull/8301) would need to be merged for this value to be propagated correctly to this location.